transcom/mymove
# MB-3148 Add createMTOAgent endpoint to Prime API

## Description

* Adds new endpoint to Prime API so MTO Agents can be added after shipment creation.
* Prime can now create RELEASING_AGENT and RECEIVING_AGENT for an mtoShipment with no existing agents.
* If the agent already exists, they'll need to use updateMTOAgent instead.


## Setup

**Tests**

```
// Tests for handler
 go test ./pkg/handlers/primeapi/ --testify.m "TestCreateMTOAgentHandler" -v

// Tests for service object
 go test ./pkg/services/mto_agent --testify.m "TestMTOAgentCreator" -v
```
**Manual Testing**

You can manually test the endpoint by making a `POST` request to `/mto-shipments/:mtoShipmentID/agents` with a similar body:
```
{
    "firstName": "Riley",
    "lastName": "Baker",
    "email": "rileybaker@example.com",
    "phone": "555-555-3020",
    "agentType": "RELEASING_AGENT"
}
```

* The shipment must be associated with an MTO that is available to Prime
* It cannot already have an agent of the same type (although worth testing to confirm you get the correct error!)


## Code Review Verification Steps

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-3148) for this change
